mlflow/store/artifact/ftp_artifact_repo.py

```diff
@@ -106,10 +106,10 @@ def list_artifacts(self, path=None):
             if not self._is_dir(ftp, list_dir):
                 return []
             artifact_files = ftp.nlst(list_dir)
-            artifact_files = list(filter(lambda x: x != "." and x != "..", artifact_files))
             # Make sure artifact_files is a list of file names because ftp.nlst
             # may return absolute paths.
             artifact_files = [os.path.basename(f) for f in artifact_files]
+            artifact_files = list(filter(lambda x: x != "." and x != "..", artifact_files))
```

Comment:
qq: Is "." malicious?

Reply:
I think we filter out `.` and `..` to avoid an infinite loop.

Reply:
Ah I see, thx for the clarification!